gms/gossiper.cc

```diff
@@ -1492,9 +1495,7 @@ void gossiper::mark_alive(inet_address addr, endpoint_state& local_state) {
             return real_mark_alive(addr, state);
         }
         return make_ready_future();
-    }).finally([this, addr] {
-        _pending_mark_alive_endpoints.erase(addr);
-    }).handle_exception([addr] (auto ep) {
+    }).handle_exception([addr, unmark_pending = std::move(unmark_pending)] (auto ep) {
```

Comment:
I think it is clear not to mix erase pending and release the background gate, e.g., 

[addr, unmark_pending = std::move(unmark_pending), gh = std::move(gh)]

Reply:
ok